src/override/malloc.cc

```diff
@@ -221,16 +297,267 @@ extern "C"
   // various events.  They are currently unused, though we may wish to reset
   // statistics on fork if built with statistics.
 
-  SNMALLOC_EXPORT void SNMALLOC_NAME_MANGLE(_malloc_prefork)(void) {}
-  SNMALLOC_EXPORT void SNMALLOC_NAME_MANGLE(_malloc_postfork)(void) {}
-  SNMALLOC_EXPORT void SNMALLOC_NAME_MANGLE(_malloc_first_thread)(void) {}
+  SNMALLOC_EXPORT SNMALLOC_USED_FUNCTION inline void _malloc_prefork(void) {}
+  SNMALLOC_EXPORT SNMALLOC_USED_FUNCTION inline void _malloc_postfork(void) {}
+  SNMALLOC_EXPORT SNMALLOC_USED_FUNCTION inline void _malloc_first_thread(void)
+  {}
+
+  /**
+   * Jemalloc API provides a way of avoiding name lookup when calling
+   * `mallctl`.  For now, always return an error.
+   */
+  int SNMALLOC_NAME_MANGLE(mallctlnametomib)(const char*, size_t*, size_t*)
+  {
+    return ENOENT;
+  }
 
+  /**
+   * Jemalloc API provides a generic entry point for various functions.  For
+   * now, this is always implemented to return an error.
+   */
+  int SNMALLOC_NAME_MANGLE(mallctlbymib)(
+    const size_t*, size_t, void*, size_t*, void*, size_t)
+  {
+    return ENOENT;
+  }
+
+  /**
+   * Jemalloc API provides a generic entry point for various functions.  For
+   * now, this is always implemented to return an error.
+   */
   SNMALLOC_EXPORT int
     SNMALLOC_NAME_MANGLE(mallctl)(const char*, void*, size_t*, void*, size_t)
   {
     return ENOENT;
   }
 
+  /**
+   * Jemalloc 3 experimental API.  Allocates  at least `size` bytes and returns
+   * the result in `*ptr`, if `rsize` is not null then writes the allocated size
+   * into `*rsize`.  `flags` controls whether the memory is zeroed and what
+   * alignment is requested.
+   */
+  int SNMALLOC_NAME_MANGLE(allocm)(
+    void** ptr, size_t* rsize, size_t size, int flags)
+  {
+    auto f = JEMallocFlags(flags);
+    size = f.aligned_size(size);
+    if (rsize != nullptr)
+    {
+      *rsize = size;
```

Comment:
Looking at the code size could still be below the size that is actually allocated by snmalloc.

snmalloc::aligned_size gives a size that when translated to a size class is guaranteed to meet the alignment, but it isn't necessarily the actual size.

```
snmalloc::aligned_size(16, 66) = 
   (66 -  1) | (16 - 1) + 1 = 80
```
But I think the next size class is 96. 
```suggestion
      *rsize = round_size(size);
```
should fix this.